sevntu-checks/src/test/java/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheckTest.java

```diff
@@ -0,0 +1,555 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2019 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.sevntu.checkstyle.checks.coding;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import com.github.sevntu.checkstyle.checks.coding.Jsr305AnnotationsCheck.NullnessAnnotation;
+import com.github.sevntu.checkstyle.checks.coding.Jsr305AnnotionsTestUtil.ExpectedWarning;
+import com.puppycrawl.tools.checkstyle.api.CheckstyleException;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.FullIdent;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+public class Jsr305AnnotationsCheckTest {
+
+    private static final String[] EXCLUDED_PACKAGES = new String[] {
+        "com.foo", "org.bar",
+    };
+
+    private static final String[] INCLUDED_PACKAGES = new String[] {
+        "com.baz", "org.hmpf",
+    };
+
+    // -------------- Function tests (execute check on test sources) --------------
+
+    @Test
+    public void testParameters() throws CheckstyleException {
+        Jsr305AnnotionsTestUtil.check(
+                new ExpectedWarning("InputJsr305AnnotationsCheckWithParameter.java", 37, 44),
+                new ExpectedWarning("InputJsr305AnnotationsCheckWithParameter.java", 42, 45),
```

Comment:
See how we write tests for checks at https://github.com/sevntu-checkstyle/sevntu.checkstyle/blob/master/sevntu-checks/src/test/java/com/github/sevntu/checkstyle/checks/coding/AvoidHidingCauseExceptionCheckTest.java#L48 .
Yours show follow the similar pattern.

Reply:
Thank you, that allowed me to get rid of some unnecessary code. 

I am however a bit surprised to now find new checkstyle warnings in the travis build that I cannot reproduce here locally (using `mvn clean verify` everything here is clear):

`[ERROR] src/test/java/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheckTest.java:[51,48] (extension) CheckstyleTestMakeup: DefaultConfiguration was not assigned by an allowed create configuration method.`

Any hint what that means? 

Am I not running the local maven build the right way? I would really like to avoid (seemingly) endless chains of new issues popping up with CI builds.

Reply:
Thank you, that helped. I have now got rid of the checkstyle violations. The checkstyle-tester output will be uploaded when it's finished.